crates/astria-sequencer/src/app.rs

```diff
@@ -523,6 +523,70 @@ mod test {
         );
     }
 
+    #[tokio::test]
+    async fn app_deliver_tx_invalid_nonce() {
```

Comment:
It's unfortunate that the test for the thing you want to test is so far away from it.

If we define request handlers as freestanding functions rather than inherent methods on types we can test the handles independently and inject the necessary state as needed.

That's how axum is designed

Reply:
Can we crate an issue around what we would need to do to fix this?

Reply:
you should be able to write a test on `UnsignedTransaction::check_stateful` directly instead of here, that would be clearer imo.

Reply:
@noot the test needs to be written against `App::deliver_tx` because it needs to protect against adding extra error layers, silently breaking the downcast_ref never activating.

Reply:
@joroshiba https://github.com/astriaorg/astria/issues/408